pkg/crc64/crc64.go

```diff
@@ -86,12 +87,13 @@ func (d *digest) Sum(in []byte) []byte {
 	return append(in, byte(s>>56), byte(s>>48), byte(s>>40), byte(s>>32), byte(s>>24), byte(s>>16), byte(s>>8), byte(s))
 }
 
-// Sum returns the MD5 checksum of the data.
-func Sum(data []byte) [Size]byte {
+// Sum returns the CRC64 checksum of the data, in given byte order.
+func Sum(data []byte, order binary.ByteOrder) [Size]byte {
```

Comment:
While I am partial to this option, it is a backwards breaking change for this API. There is however a middle ground, which is to use the options pattern here, with no option being set defaulting to the original behaviour.

Reply:
Ah, I didn't think of this as a public API. I'll look into an option.

Reply:
Could I hang it off `digest` and implement the option for both `New` and `Sum`? I feel like that would be nice and symmetrical.

Reply:
`digest` is implementing [`hash.Hash`](https://github.com/golang/go/blob/master/src/hash/hash.go#L26) so they would need to be different functions to maintain this implementation, like `NewWithByteOrder`, etc.

Reply:
I don't think that's necessarily a blocker, it should still implement `hash.Hash`. But maybe I'm missing something. I'll give it a shot tomorrow.

Reply:
I had something like this in mind: https://github.com/hamba/avro/pull/491/commits/f47b59b422a96bc1cc97d3cf586728ed1bbb5bbb

Reply:
I received some out-of-band feedback that options pattern is a bit annoying, and I have a feeling `NewWithByteOrder` and `SumWithByteOrder` would be clearer. If we can assume we won't need more options later, of course.

Reply:
The options pattern allows for expansion that is non-breaking, which can in cases be very valuable. But as with all things there are tradeoffs. I do not see much expansion in this case, so the specific methods would work fine.

Reply:
Updated. Let me know what you had in mind for the README.